metapkg/packages/rust.py

```diff
@@ -41,9 +42,14 @@ def get_build_install_script(self, build) -> str:
             target = "--target x86_64-unknown-linux-musl"
         else:
             target = ""
+        if platform.system() == "Darwin":
+            sed = "gsed"
```

Comment:
Let's use the general tool abstraction.  Do `self._system_tools["sed"] = "gsed"` in `macos.Build.preapre()` and then use `sh_get_command('sed')` instead of hardcoding `sed` in scripts.